src/compiler/crystal/semantic/ast.cr

```diff
@@ -139,7 +139,7 @@ module Crystal
     property? new = false
 
     # Annotations on this def
-    property annotations : Hash(AnnotationType, Annotation)?
+    property annotations : Hash(AnnotationType, Array(Annotation))?
```

Comment:
Hmm, I can't necessarily come up with a use case for a type to have 2 _identical_ annotations, so that might not be a bad idea.

Can see what the others think.

Reply:
```crystal
@[Link("foo")]
@[Link("bar")]
```

Reply:
@ysbaddaden What about it?

Reply:
I think it might make sense to preserve the order of the applied annotations. So `Array` seems good IMO.

Reply:
@bcardiff `Set` doesn't do it?

Reply:
@Sija No, because a `Set` implements a collection of unordered values with no duplicates

Reply:
So should I be looking into switching it to a `Set` then?  What would be you're ideas to return the last element since `Set` doesn't implement a `last?` method?

Reply:
I think `Array` is fine.